internal/engine/experiment/tlsmiddlebox/tracing_test.go

```diff
@@ -0,0 +1,249 @@
+package tlsmiddlebox
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/ooni/probe-cli/v3/internal/model"
+	"github.com/ooni/probe-cli/v3/internal/netxlite"
+	"github.com/ooni/probe-cli/v3/internal/netxlite/filtering"
+)
+
+func TestIterativeTrace(t *testing.T) {
+	t.Run("on success", func(t *testing.T) {
+		server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I would measure how much time it takes for each test to run and I think you should apply this pattern

```
if testing.Short() {
  t.Skip("skip test in short mode")
}
```

This (including the same exact string) is how we make `go test -short` faster.

When you have multiple `t.Run` inside a test, I suggest to apply this pattern only to the long `t.Run`s.